app/javascript/packs/history.jsx

```diff
@@ -0,0 +1,22 @@
+import { h, render } from 'preact';
+import { getUserDataAndCsrfToken } from '../chat/util';
+import { History } from '../history/history';
+
+function loadComponent() {
+  getUserDataAndCsrfToken().then(({ currentUser }) => {
```

Comment:
I think it makes more sense to get the user data before, instead of using a `setInterval` function like in the reading list. This way the component just accepts the availableTags from the outside and that's it